src/plugins/data_views/server/data_views_service_factory.ts

```diff
@@ -43,14 +44,16 @@ export const dataViewsServiceFactory = (deps: DataViewsServiceFactoryDeps) =>
     const { logger, uiSettings, fieldFormats, capabilities, rollupsEnabled } = deps;
     const uiSettingsClient = uiSettings.asScopedToClient(savedObjectsClient);
     const formats = await fieldFormats.fieldFormatServiceFactory(uiSettingsClient);
+    const excludedTiers = await uiSettingsClient.get<string>(DATA_VIEWS_FIELDS_EXCLUDED_TIERS);
```

Comment:
Would it become necessary to restart Kibana after editing the setting?

Reply:
I don't think so but I'd appreciate your verification - `dataViewsServiceFactory` is a function that returns a function and is made available here - https://github.com/elastic/kibana/blob/main/src/plugins/data_views/server/plugin.ts#L10

Reply:
So it's called on plugin start that means that at least a reload in the browser would be required, right? We could let users know about that via additional `requiresPageReload: true` prop for `data_views:fields_excluded_data_tiers` Advanced Setting.

I am not sure how to test but my guess was that if some other code (maybe alerts?) uses only server side APIs for data views, they might not get updates after the setting is changed.

Reply:
>So it's called on plugin start

That provides the dependencies, returning another function which is called when a Request object is available, such as here - https://github.com/elastic/kibana/blob/main/src/plugins/data_views/server/rest_api_routes/public/get_data_view.ts#L85

No browser restart is needed since the bulk of the function is executed during the request.